session/session.go

```diff
@@ -2215,6 +2215,15 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 		recordSet, err = runStmt(ctx, s, stmt)
 	}
 
+	// Observe the resource group query total counter if the resource control is enabled and the
+	// current session is attached with a resource group.
+	if variable.EnableResourceControl.Load() && !s.isInternal() {
```

Comment:
What about only check `len(resourceGroupName) != 0` here which means the resource group is explicitly set by user so we can support explicitly set resource group to the default group, and maybe at some time the internal session should also bind to a group that should be tracked.